libraries/chain/authorization_manager.cpp

```diff
@@ -559,7 +559,7 @@ namespace eosio { namespace chain {
 
       }
 
-      if( !allow_unused_keys  || check_but_dont_fail) {
+      if( !allow_unused_keys  && !check_but_dont_fail) {
          EOS_ASSERT( checker.all_keys_used(), tx_irrelevant_sig,
```

Comment:
Instead, I think the `EOS_ASSERT` should be modified, since we would ideally want to include the time it takes for `checker.all_keys_used()`

```
      if( !allow_unused_keys ) {
         EOS_ASSERT( checker.all_keys_used() || check_but_dont_fail, tx_irrelevant_sig,
```